src/MessageList/MessageListItem.vala

```diff
@@ -124,11 +130,33 @@ public class Mail.MessageListItem : Gtk.ListBoxRow {
         }
 
         add (base_grid);
-        show_all ();
 
         destroy.connect (() => {
             loading_cancellable.cancel ();
         });
+
+        var sender = message_info.from;
```

Comment:
As discussed in slack, you need to do the following to get just the email address:
```
string from_address;
message.get_from ().@get (0, null, out from_address);
```

Although, you don't have access to the message reference here I don't think.